### PR TITLE
Sync fix

### PR DESCRIPTION
**Pull request by ObsessiveOrange**

### CODE REVIEW

File: src/main/java/cceclipseplugin/editor/DocumentManager.java [hunk 1]
```diff
@@ -117,8 +117,11 @@ public ITextEditor getEditor(String absolutePath) {
 	 *
 	 * @return The queue of diffs that was applied.
```
Comment (fahslaj) [inline src/main/java/cceclipseplugin/editor/DocumentManager.java]: You removed the Queue, so is this JavaDoc looks to be out of date.

File: src/main/java/cceclipseplugin/editor/DocumentManager.java [hunk 2]
```diff
@@ -174,16 +177,16 @@ public void handleNotification(Notification n) {
...
+//		if (changeNotif.fileVersion <= fileMeta.getVersion()) {
+//			try {
+//				System.out.printf(
+//						"ChangeNotification version was less than or equal to current version: %d <= %d; Notification: ",
+//						changeNotif.fileVersion, fileMeta.getVersion(), new ObjectMapper().writeValueAsString(n));
+//			} catch (JsonProcessingException e) {
+//				e.printStackTrace();
+//			}
+//			return;
+//		}
```
Comment (fahslaj) [inline src/main/java/cceclipseplugin/editor/DocumentManager.java]: Why is all of this code commented out? Where did this logic go?

File: src/main/java/cceclipseplugin/editor/listeners/DocumentChangeListener.java [hunk 3]
```diff
@@ -82,12 +83,17 @@ public void documentAboutToBeChanged(DocumentEvent event) {
...
+			synchronized(docMgr.getAppliedDiffs(currFile)){
+				for(int j = 0; j < docMgr.getAppliedDiffs(currFile).size(); j++){
+					Diff appliedDiff = docMgr.getAppliedDiffs(currFile).get(j);
+					System.out.printf("DEBUG SEND-ON-NOTIF: %s ?= %s; %b\n", diffs.get(i).toString(),
+							appliedDiff.toString(), diffs.get(i).equals(appliedDiff));
+					if(appliedDiff.equals(diffs.get(i))){
+						for(int k = j-1; k >= 0; k--){
+							docMgr.getAppliedDiffs(currFile).removeFirst();
+						}
+						continue diffLoop;
```
Comment (fahslaj) [inline src/main/java/cceclipseplugin/editor/listeners/DocumentChangeListener.java]: `docMgr.getAppliedDiffs` is called 4 times in the last 10 lines. Capture the applied diffs in a local variable right before the synchronized block.

**Review by fahslaj — CHANGES_REQUESTED** [review]

File: src/main/java/cceclipseplugin/editor/DocumentManager.java (hunk 2 above)
Comment (ObsessiveOrange) [reply src/main/java/cceclipseplugin/editor/DocumentManager.java]: That logic was outdated; I'm commenting it out as a reminder to build the patch reordering. Right now we're simply assuming it comes in order.

File: src/main/java/cceclipseplugin/editor/listeners/DocumentChangeListener.java (hunk 3 above)
Comment (ObsessiveOrange) [reply src/main/java/cceclipseplugin/editor/listeners/DocumentChangeListener.java]: done

File: src/main/java/cceclipseplugin/editor/DocumentManager.java (hunk 1 above)
Comment (ObsessiveOrange) [reply src/main/java/cceclipseplugin/editor/DocumentManager.java]: Not really. I've put it into a hashmap, and used a linkedlist instead, but in all reality, it's a queue. I just needed access to all the methods of the linkedlist.

**Review by fahslaj — DISMISSED** [review]

File: src/main/java/cceclipseplugin/core/PluginManager.java [hunk 4]
```diff
@@ -77,7 +77,7 @@
 
 	// PLUGIN SETTINGS
 	// TODO: move to configuration file
-	final private String WS_ADDRESS = "wss://codecollaborate.obsessiveorange.com:8000/ws/";
+	final private String WS_ADDRESS = "ws://cody.csse.rose-hulman.edu:8000/ws/";
```
Comment (julie-is-late) [inline src/main/java/cceclipseplugin/core/PluginManager.java]: revert this back to use prod

**Review by julie-is-late — APPROVED** [review]

**Pull request merged**